File: number_processor.c
#include <stdio.h>

void processNumber(int num) {
    if (num == 1) {
        printf("%d\n", num);
        return;
    }
    
    printf("%d -> ", num);

    if (num % 2 != 0) {
        processNumber(num * 3 + 1);
    } else {
        processNumber(num / 2);
    }
}

int main() {
    int num;
    
    printf("Enter a number between 1 and 9: ");
    scanf("%d", &num);
    
    if (num < 1 || num > 9) {
        printf("Invalid input. Please enter a number between 1 and 9.\n");
        return 1;
    }

    processNumber(num);

    return 0;
}

File: number_processor.py
def process_number(num):
    if num == 1:
        print(num)
        return
    
    print(num, "->", end=" ")

    if num % 2 != 0:
        process_number(num * 3 + 1)
    else:
        process_number(num // 2)

num = int(input("Enter a number between 1 and 9: "))

if num < 1 or num > 9:
    print("Invalid input. Please enter a number between 1 and 9.")
else:
    process_number(num)
